### DISCUSSION TITLE
Why don't you change online status when online / offline event occurs?

### DISCUSSION THREAD

**Discussion by kimsupin** [TanStack/query#5660] [Ideas]
[related to this discussion](https://github.com/TanStack/query/discussions/5587)

I ask something about listening to the online / offline events.

In onlineManager.ts, The setup method of the OnlineManager class puts a callback function that receives a boolean value of online as a parameter when online / offline events of the window occur. But the callback function doesn't actually receive the online value as a parameter, so when does this.setOnline(online) statement get executed? It seems that only the this.onOnline() statement will always be executed.

For example, if onlineManager.setOnline(false) is executed 3 seconds after entering a web page where the network is online, and then the network is disconnected and reconnected, the online event will occur, but the online value of onlineManager is still false, so the api call will not run.

Isn't it correct to change onlineManager's setup function as follows?
```ts
this.setup = (onOnline) => {
      if (!isServer && window.addEventListener) {
        const listener = (event: 'online' | 'offline') => {
            this.setOnline(event === 'online') // it is needed, isnt' it?
            onOnline()
        }
        // Listen to online
        onlineEvents.forEach((event) => {
          window.addEventListener(event, listener, false)
        })

        return () => {
          // Be sure to unsubscribe if a new handler is set
          onlineEvents.forEach((event) => {
            window.removeEventListener(event, listener)
          })
        }
      }

      return
    }

```

**Comment by TkDodo** [comment]
> For example, if onlineManager.setOnline(false) is executed 3 seconds after entering a web page where the network is online, and then the network is disconnected and reconnected, the online event will occur, but the online value of onlineManager is still false, so the api call will not run.

if you call `onlineManager.setOnline(false)`, you've opted into manually managing the online/offline behaviour. If you want to go back to the default setting, you have to call `onlineManager.setOnline(undefined)`
